fix(dev-guide): updated contributing guideline (#506)
jitsi/handbook: docs/dev-guide/contributing.md

title: Contributing Guidelines
---

# 🤝 How to Contribute 
We greatly appreciate your willingness to contribute ❤️ Before you start working however, please take a moment to read and follow this brief guide.

# 📥 Reporting Issues and Asking Questions 

> - We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).

Reviewer: Shall we re-discuss this @damencho ? I think it's not a very natural flow. People find the issue tracker first. Then we send them away, but the issue might come back once confirmed.

That creates some friction.

I think improviing the issue template / questionaire would help, and once an issue is identified as something that should be treated in the forum we can close it as such. Thoughts?

Reviewer 2: I suggest moving forward with this PR as it's an improvement and describes the current procedure. Let's re-discuss separately.

Reviewer: Good point. I'll approve as I have no other comments.

title: Contributing Guidelines
---

# How to Contribute 🤝
We greatly appreciate your willingness to contribute. Before you start working however, please take a moment to read and follow this brief guide.

# Reporting Issues and Asking Questions 💭

We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).

Feel free to report ***any bugs, ask for new features or anything else*** you need help with. When opening an issue, please provide as much information as possible.

> For bugs please follow these steps:

Reviewer 3: We ask people to report and discuss problems in the community forum, and when we think that is an issue with the code we ask them to open a ticket in the relevant repo. 

Many times they create tickets which we close as they are in wrong repo, or that is not a code problem, but configuration problems, or those are issues they have with meet.jit.si and so on.
This step is important and should not be removed.

Author: I have this sentence that explains it. But if it is not very clear, I can add back what you sent

<img width="1208" alt="Screenshot 2024-05-02 at 14 41 32" src="https://github.com/jitsi/handbook/assets/43909097/479c3f60-8e20-43c0-971f-5958bba900bf">

Reviewer 3: The first step of reporting an issue is opening a discussion in the forum.

Author: Okay I edited it and made a few more edits

title: Contributing Guidelines
---

# 🤝 How to Contribute 
We greatly appreciate your willingness to contribute ❤️ Before you start working however, please take a moment to read and follow this brief guide.

# 📥 Reporting Issues and Asking Questions 

> - We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the [issue tracker](https://github.com/jitsi/jitsi-meet/issues).

Reviewer 2: ```suggestion
- We prefer issues to be discussed first in the [community forum](https://community.jitsi.org/) and when confirmed, then an issue can be opened in the issue tracker of the appropriate project on GitHub.
```

- Feel free to report ***any bugs, ask for new features or anything else*** you need help with. When opening an issue, please provide as much information as possible.

- Please ask any general and implementation specific questions on the [community forum](https://community.jitsi.org/) for support.

### 🪲 Bug Reports and Other Issues

For bugs please follow these steps:

- **Provide Detailed Information:**
  Include versions of Jitsi Meet, Jicofo, and JVB.

- **Description of the Issue:**
  Clearly explain the problem encountered.

- **Reproduction Steps:**
  Provide step-by-step instructions to recreate the issue.

- **Expected Behavior:**
  Describe the expected outcome when using the software.

- **Actual Behavior:**
  Explain what actually happened, including any error messages.

### 💟 Feature Requests

> If you have an idea for a new feature or something you'd like to see improved in the Jitsi, here's how you can let us know:

Reviewer 2: ```suggestion
If you have an idea for a new feature or something you'd like to see improved in Jitsi, here's how you can let us know:
```

### Kotlin

- For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.

## ✅ Code Reviews

- **Submit Your Contribution:** After completing your work, submit your contribution.

> - **Automatic Assignment:** Your pull request will be automatically assigned to appropriate reviewers based on the affected codebase and expertise required for the changes.

Reviewer 2: I don't think we have that setup. Unless it's something automagical from github?

- **Review Process:** Reviewers will carefully examine your code, checking for adherence to coding standards, correctness, performance and potential issues.
- **Feedback and Iteration:** If any issues or suggestions are identified during the review, you'll receive feedback from the reviewers. Address any comments or concerns raised and make necessary revisions to your code.

> - **Test Your Changes:** Before finalizing your contribution, ensure that any relevant tests are run to verify the correctness and functionality of your changes.

Reviewer 2: Relevant testing and linting should be done locally before even opening a PR. 

```suggestion
- **Automated tests:** Once the PR is in a good state, a team member will trigger the automated tests. The PR needs to merge cleanly on top of master, and test failures or issues discovered at this stage will need to be addressed before the PR is approved for merging.
```

Author: Thanks for your comments, all fixed @bgrozev

### Kotlin

- For Kotlin code we use the [standard convention](https://kotlinlang.org/docs/coding-conventions.html) and limit line length to 120 characters. We use `ktlint` to enforce formatting.

## ✅ Code Reviews

> - **Submit Your Contribution:** After completing your work, submit your contribution.

Reviewer: Let's add some text to also encourage Draft PRs if an approach should be discussed. It's frustrating to finish some work only to scrap it later. Better to open a draft PR early for discussion.

> - **Automatic Assignment:** Your pull request will be automatically assigned to appropriate reviewers based on the affected codebase and expertise required for the changes.

Reviewer: This is not the case.

Author: I already fixed that
